src/main/java/io/jenkins/plugins/pipeline_elasticsearch_logs/write/fluentd/FluentdWriter.java

```diff
@@ -136,87 +154,94 @@ public String getDisplayName() {
     }
 
     /**
-     * Logs the given object to Fluentd. This method blocks if Fluentd is not reachable and
-     * throws an IOException after TIMEOUT_MILLIS.
+     * Logs the given object to Fluentd asynchronously
+     * Checks if there have been any data that couldn't be sent after several retries and then fails.
      *
      * @param data The data to post
      * @throws IOException if something failed permanently
      */
     @Override
     public void push(Map<String, Object> data) throws IOException {
-        if(fluentd == null || configChanged) initFluentdLogger();
-        try {
-
-            long timeMarker = System.currentTimeMillis();
-            boolean sent = fluentd.log(tag, data);
-            if(!sent) {
-                //Errors are not reported explicitly. But the latest error should be the reason for (sent == false)
-                Exception lastError = errorHandler.getLastError(timeMarker);
-                throw new IOException("Could not send logs to fluentd", lastError);
+        if(fluentd == null) initFluentdLogger();
```

Comment:
If the config changes still the existing fluentd (with old values) is used. Not an issue in the JFR case, but if someone would use the plugin in a static Jenkins.
On which granularity are FluentdWriter instances created? If it is per build (or more often) it is fine I guess.

Reply:
They are created per build step basically.